temporal-sdk/src/main/java/io/temporal/internal/sync/POJOActivityTaskHandler.java

```diff
@@ -110,18 +109,15 @@ private void registerActivityImplementation(
     }
     Class<?> cls = activity.getClass();
     POJOActivityImplMetadata activityImplMetadata = POJOActivityImplMetadata.newInstance(cls);
-    for (POJOActivityInterfaceMetadata activityInterface :
-        activityImplMetadata.getActivityInterfaces()) {
-      for (POJOActivityMethodMetadata activityMetadata : activityInterface.getMethodsMetadata()) {
-        String typeName = activityMetadata.getActivityTypeName();
-        if (activities.containsKey(typeName)) {
-          throw new IllegalArgumentException(
-              "\"" + typeName + "\" activity type is already registered with the worker");
-        }
-        Method method = activityMetadata.getMethod();
-        ActivityTaskExecutor implementation = newTaskExecutor.apply(method, activity);
-        activities.put(typeName, implementation);
+    for (POJOActivityMethodMetadata activityMetadata : activityImplMetadata.getActivityMethods()) {
```

Comment:
I had to make this change because otherwise if inheritance exists in activity interfaces, we end up getting duplication method metadata causing IllegalArgumentException (https://buildkite.com/temporal/java-sdk-public/builds/246#84704748-21be-4519-981e-d8cfe8c91e2e). `activityImplMetadata.getActivityMethods()` returns de-duped methods